lib/datadog/appsec/contrib/graphql/gateway/watcher.rb

```diff
@@ -24,27 +24,30 @@ def watch_multiplex(gateway = Instrumentation.gateway)
                 gateway.watch('graphql.multiplex', :appsec) do |stack, gateway_multiplex|
                   block = false
                   event = nil
+
                   scope = AppSec::Scope.active_scope
 
-                  AppSec::Reactive::Operation.new('graphql.multiplex') do |op|
-                    GraphQL::Reactive::Multiplex.subscribe(op, scope.processor_context) do |result|
-                      event = {
-                        waf_result: result,
-                        trace: scope.trace,
-                        span: scope.service_entry_span,
-                        multiplex: gateway_multiplex,
-                        actions: result.actions
-                      }
+                  if scope
```

Comment:
is it possible to use `return nil unless scope` here instead of wrapping it with `if`

Reply:
yes, this would actually be better. Fixed here: [31da3fb](https://github.com/DataDog/dd-trace-rb/pull/3887/commits/31da3fba8f87e39277387726e4b4d1526d02bf35)

Reply:
now I'm having second thoughts about my comment 😓 

Is it ok that when we use `return` the following will not be called:
```ruby
                  next [nil, [[:block, event]]] if block
                  ret, res = stack.call(gateway_multiplex.arguments)
                  if event
                    res ||= []
                    res << [:monitor, event]
                  end
                  [ret, res]
```

I might have proposed some premature optimisation

Reply:
I think this is fine considering that `event` is set within [`GraphQL::Reactive::Multiplex.subscribe` block](https://github.com/DataDog/dd-trace-rb/blob/35f8ccd7703337dfbe333d9af038a7f68c3bf4f0/lib/datadog/appsec/contrib/graphql/gateway/watcher.rb#L32) that relies on the active scope being present.

Reply:
no, actually this is a block that is passed to `watch`, so we should use `next` with appropriate arguments, not `return`

I confused you previously, I think the version with `if scope` was actually more correct

Reply:
yes, you are right, I removed my commit